### PR TITLE
create_changelog: Filter out already reported items

### PR DESCRIPTION
**Pull request by lanzagar**
##### Issue

Changes for a new release were repeating items from previous bugfix releases.

##### Description of changes
Grab all PR ids (#XXXX) from existing changelog and filter them out from the proposed new ones.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: scripts/create_changelog.sh [hunk 1]
```diff
@@ -1,16 +1,20 @@
 #!/bin/bash
+set -u -e -o pipefail
 
 VERSION=$(grep '^VERSION' setup.py | sed -E "s/.*'([^']*)'/\1/")
 RELEASE_DATE=$(date "+%Y-%m-%d")
 
+# compile a list of already reported PRs
+grep -E "#[0-9]{4}" -o CHANGELOG.md | sort > /tmp/reported
```
Comment (astaric) [inline scripts/create_changelog.sh]: You could use mktemp to create a temporary file.

Comment (lanzagar) [inline scripts/create_changelog.sh]: Yes, much better.

File: scripts/create_changelog.sh [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+tempfile="$(mktemp)"
+grep -E "#[0-9]{4}" -o CHANGELOG.md | sort > "$tempfile"
+
 echo "[$VERSION] - $RELEASE_DATE"
 echo "--------------------"
 echo '##### Enhancements'
 git log stable..master --first-parent --format='%s %b' |
 sed -E 's/.*#([0-9]+).*\[ENH\] *(.*)/\* \2 ([#\1](\.\.\/\.\.\/pull\/\1))/' |
-    grep -E '^\*'
+    grep -E '^\*' | grep -v -F -f "$tempfile"
```
Comment (astaric) [inline scripts/create_changelog.sh]: Should this go into a new line?

File: scripts/create_changelog.sh [hunk 3]
```diff
@@ -1,16 +1,21 @@
 #!/bin/bash
+set -u -e -o pipefail
 
 VERSION=$(grep '^VERSION' setup.py | sed -E "s/.*'([^']*)'/\1/")
 RELEASE_DATE=$(date "+%Y-%m-%d")
 
+# compile a list of already reported PRs
+tempfile="$(mktemp)"
```
Comment (astaric) [inline scripts/create_changelog.sh]: Is there a better name that could be used? As /tmp/reported was used before, maybe "reported"?

File: scripts/create_changelog.sh (hunk 2 above)
Comment (lanzagar) [inline scripts/create_changelog.sh]: I don't see a reason for a new line, but if there are some bash style guidelines that someone prefers we can use them.

A better improvement would be to indent the line above.

File: scripts/create_changelog.sh (hunk 3 above)
Comment (lanzagar) [inline scripts/create_changelog.sh]: Better name for the variable or for the file?
I can change it if you have a suggestion.
`reported="$(mktemp)"` ?

File: scripts/create_changelog.sh [hunk 4]
```diff
@@ -1,16 +1,21 @@
 #!/bin/bash
+set -u -e -o pipefail
 
 VERSION=$(grep '^VERSION' setup.py | sed -E "s/.*'([^']*)'/\1/")
 RELEASE_DATE=$(date "+%Y-%m-%d")
 
+# compile a list of already reported PRs
+reported="$(mktemp)"
+grep -E "#[0-9]{4}" -o CHANGELOG.md | sort > "$tempfile"
```
Comment (kernc) [inline scripts/create_changelog.sh]: Where is `tempfile` assigned?

**Pull request merged**